app/discovery/discovery.go

```diff
@@ -146,6 +148,37 @@ func (s *Service) Match(srv, src string) (string, MatchType, bool) {
 	return src, MTProxy, false
 }
 
+// ScheduleHealthCheck starts background loop with health-check
+func (s *Service) ScheduleHealthCheck(ctx context.Context, interval time.Duration) {
+	log.Printf("health-check scheduled every %s", interval)
+
+	go func() {
+	hloop:
```

Comment:
I don't see why label even needed here, just return?

Reply:
Isn't it going for a new iteration in the loop after return in select? I'd like to leave the method after context is done.

Reply:
not sure what you mean, I thought of return in ctx case which will do the same thing, i.e. finish the goroutine